src/App.jsx

```diff
@@ -7,27 +7,67 @@ import RestaurantList from "./components/Main/RestaurantList.jsx"
 import Restaurants from "./data/Restaurant.js";
 import { useState } from "react";
 
-function App() { 
+function App() {
+  const [ restaurantList, setRestaurantList ] = useState(Restaurants());
   const [ category, setCategory ] = useState("전체");
-  const restaurantList = Restaurants();
+  const [ clickRestaurantItem, setClickRestaurantItem ] = useState(null);
+  const [isModalOpen, setIsModalOpen] = useState({
+    detail: false,
+    add: false,
+  });
 
   const filteredRestaurants = category === "전체" ? restaurantList : restaurantList.filter(
     (restaurant) => restaurant.category === category
   );
+  
+  const selectedRestaurant = restaurantList.find(
+      (restaurant) => restaurant.name === clickRestaurantItem
+  )
+
+  const openDetailModal = (restaurantName) => {
+    setClickRestaurantItem(restaurantName);
+    setIsModalOpen((prev) => ({ ...prev, detail: true }));
+  };
+
+  const closeDetailModal = () => {
+    setIsModalOpen((prev) => ({ ...prev, detail: false }));
+  }
+
+  const openAddModal = () => {
+    setIsModalOpen((prev) => ({ ...prev, add: true }));
+  }
+
+  const closeAddModal = (event) => {
+    if(event !== null)
+      addRestaurant(event);
+    setIsModalOpen((prev) => ({ ...prev, add: false }));
+  }
 
   return (
     <>
-      <Header />
+      <Header onChangeAddModal={openAddModal}/>
```

Comment:
react에 이벤트가 onClick, onChange, onSubmit 이렇게 있어서 
함수 명을 전달할 때 앞에 on을 붙이는 것이 일반적인데 이 경우는
header를 `click`하는 이벤트를 전달하는 것이니
```suggestion
      <Header onClickAddModal={openAddModal}/>
```
이렇게 바꾸면 클릭하는 이벤트를 실행할 것이다 라고 알아채기 쉬울 거 같아요! 
그리고 이벤트를 처리하는 함수에는 handle을 붙이는 것이 일반적이라고 합니다!

Reply:
이것도 네이밍을 과거로 하는게 좋을까요?? `onClickedAddModal` 이렇게요!